resources/timerbox.ts

```diff
@@ -416,6 +431,12 @@ export default class TimerBox extends HTMLElement {
   advance(): void {
     const elapsedSec = (new Date().getTime() - this._start) / 1000;
     if (elapsedSec >= this._duration) {
+      // We need to check for this._duration > 0 here, as for undocumented reason the
+      // duration of a timerbox is always set to zero before it is set to the
+      // actual duration. As a result this would otherwise trigger a sound each time
+      // the ability is activated.
+      if (this._duration > 0 && this._playSoundNotification && this._soundNotification === 'expired')
+        this.playNotification();
```

Comment:
I don't really like adding this to the timerbox directly and was wondering if adding a callback/event mechanism that fires when the threshold is reached or the timer expired would be a better alternative. This would allow attaching different actions to different timerboxes and would also remove the requirement to pass all kinds of configuration options into the timerbox.

Reply:
I agree.  If possible, I think a callback would be nicer here rather than baking sounds inside of the web component.  Is it possible to do that in this PR?

Reply:
Yes, I'll change it to support callbacks for the different relevant states.